src/api/dependency-graph/src/graph-collector/index.js

```diff
@@ -0,0 +1,136 @@
+const yaml = require('js-yaml');
+const fs = require('fs');
+const { logger, fetch } = require('@senecacdot/satellite');
+const Graph = require('./graph');
+
+function extractRepositoryUrl(repository) {
+  let possibleRepoLink = repository['url'] || repository;
+
+  if (possibleRepoLink.startsWith('git+')) {
+    possibleRepoLink = possibleRepoLink.substring(4);
+  }
+
+  try {
+    const url = new URL(possibleRepoLink);
+    return possibleRepoLink;
+  } catch (error) {
+    logger.warn({ possibleRepoLink }, 'Could not extract repository link');
+  }
+
+  return null;
+}
+
+function extractVersionAndPeerDependency(versionString) {
+  const versionRegex = /^(\d+\.\d+\.\d+)_(.+)$/;
+
+  const matches = versionString.match(versionRegex);
+
+  if (matches) {
+    const version = matches[1];
+    const peerDependencyRegex = /^(.+)@(\d+\.\d+\.\d+)$/;
+    const peerDependency = matches[2].match(peerDependencyRegex);
+    const peerDepName = peerDependency[1].replace('+', '/');
+    const peerDepVersion = peerDependency[2];
+
+    return {
+      version,
+      peerDependency: {
+        [peerDepName]: peerDepVersion,
+      },
+    };
+  }
+
+  return { version: versionString, peerDependency: {} };
+}
+
+function extractInfoFromProjectPath(projectPath) {
+  const pathComponents = projectPath.substring(1).split('/');
+  let projectName;
+
+  if (pathComponents.length === 2) {
+    projectName = pathComponents[0];
+  } else {
+    const namespace = pathComponents[0];
+    const packageName = pathComponents[1];
+
+    projectName = `${namespace}/${packageName}`;
+  }
+
+  const extractedValues = extractVersionAndPeerDependency(projectVersionWithPossiblePeer);
+
+  return {
+    projectName,
+    projectVersion: extractedValues.version,
+    peerDependency: extractedValues.peerDependency,
+  };
+}
+
+function processDependencyObject(dependenciesObj) {
+  Object.keys(dependenciesObj).forEach((key) => {
+    const possibleVersionWithPeer = dependenciesObj[key];
+    const { version } = extractVersionAndPeerDependency(possibleVersionWithPeer);
+    dependenciesObj[key] = version;
+  });
+}
+
+async function createGraphFromSources() {
+  const graph = new Graph();
+
+  const pnpmLock = yaml.load(
```

Comment:
Let's not do this.  @DukeManh's idea of calling `pnpm list` in a child-process, and having it return JSON is probably the better route.